<launch>
  <arg name="path_frame" default="base_link"/>
  <arg name="cmd_frame" default="base_link"/>
  <arg name="max_speed" default="0.45"/>
  <arg name="speed_gain" default="0.5"/>
  <arg name="lookahead_distance" default="0.5"/>
  <arg name="goal_tolerance" default="0.2"/>
  <arg name="angular_gain" default="2.0"/>
  <arg name="max_yaw_rate" default="1.0"/>
  <arg name="safety_radius" default="0.45"/>
  <arg name="max_path_length" default="4.0"/>
  <arg name="goal_offset" default="1.0"/>
  <arg name="publish_rate_hz" default="15.0"/>
  <arg name="goal_tf_timeout" default="0.5"/>
  <arg name="is_real_robot" default="true"/>
  <arg name="scan_topic" default="/utlidar/cloud"/>
  <arg name="transformed_scan_topic" default="/utlidar/transformed_cloud"/>
  <arg name="goal_tf_frame" default="suitcase_frame"/>
  <arg name="obstacle_z_min" default="-0.0"/>
  <arg name="obstacle_z_max" default="1.0"/>
  <arg name="grid_resolution" default="0.2"/>
  <arg name="max_considered_range" default="4.0"/>
  <arg name="lidar_frame" default="utlidar_lidar"/>
  <arg name="lidar_x" default="0.0"/>
  <arg name="lidar_y" default="0.0"/>
  <arg name="lidar_z" default="0.0"/>
  <arg name="lidar_roll" default="0.0"/>
  <arg name="lidar_pitch" default="2.8782025850555556"/>
  <arg name="lidar_yaw" default="0.0"/>

  <node pkg="local_lidar_planner" exec="utlidar_transformer.py" name="utlidar_transformer" output="screen">
    <param name="input_topic" value="$(var scan_topic)"/>
    <param name="output_topic" value="$(var transformed_scan_topic)"/>
    <param name="target_frame" value="$(var path_frame)"/>
    <param name="lidar_frame" value="$(var lidar_frame)"/>
    <param name="lidar_x" value="$(var lidar_x)"/>
    <param name="lidar_y" value="$(var lidar_y)"/>
    <param name="lidar_z" value="$(var lidar_z)"/>
    <param name="lidar_roll" value="$(var lidar_roll)"/>
    <param name="lidar_pitch" value="$(var lidar_pitch)"/>
    <param name="lidar_yaw" value="$(var lidar_yaw)"/>
  </node>

  <node pkg="local_lidar_planner" exec="local_lidar_planner_simple.py" name="local_lidar_planner_simple" output="screen">
    <param name="path_frame" value="$(var path_frame)"/>
    <param name="max_path_length" value="$(var max_path_length)"/>
    <param name="publish_rate_hz" value="$(var publish_rate_hz)"/>
    <param name="safety_radius" value="$(var safety_radius)"/>
    <param name="scan_topic" value="$(var transformed_scan_topic)"/>
    <param name="goal_tf_frame" value="$(var goal_tf_frame)"/>
    <param name="goal_offset" value="$(var goal_offset)"/>
    <param name="goal_tf_timeout" value="$(var goal_tf_timeout)"/>
    <param name="obstacle_z_min" value="$(var obstacle_z_min)"/>
    <param name="obstacle_z_max" value="$(var obstacle_z_max)"/>
    <param name="grid_resolution" value="$(var grid_resolution)"/>
    <param name="max_considered_range" value="$(var max_considered_range)"/>
  </node>

  <node pkg="local_lidar_planner" exec="path_follower_simple.py" name="path_follower_simple" output="screen">
    <param name="path_frame" value="$(var path_frame)"/>
    <param name="cmd_frame" value="$(var cmd_frame)"/>
    <param name="max_speed" value="$(var max_speed)"/>
    <param name="speed_gain" value="$(var speed_gain)"/>
    <param name="lookahead_distance" value="$(var lookahead_distance)"/>
    <param name="goal_tolerance" value="$(var goal_tolerance)"/>
    <param name="angular_gain" value="$(var angular_gain)"/>
    <param name="max_yaw_rate" value="$(var max_yaw_rate)"/>
    <param name="is_real_robot" value="$(var is_real_robot)"/>
  </node>

</launch>
